### ISSUE TITLE
TypeError when using jax.vjp with jnp.take_along_axis

### ISSUE THREAD

**Issue by shaobohou** [google/jax#11256]
This can be reproduced on python 3.9.12, but not 3.8 or 3.7.

```python
def func(xs, ys):
  return jnp.take_along_axis(xs, ys[:, jnp.newaxis], axis=-1)

features = np.array([np.arange(5), np.arange(5, 0, -1)], dtype=np.float32)
labels = np.array([2, 3])

# Fails with TypeError: a bytes-like object is required, not 'int'                                                                                                                                                                                      
vjp_outputs, vjp_fn = jax.vjp(func, features, labels)
```

The vjp call fails with 

```python
  File "/tmp/copybara_output/tf2jax/_src/blah.py", line 33, in func
    return jnp.take_along_axis(xs, ys[:, jnp.newaxis], axis=-1)
  File "/tmp/tf2jax_test/lib/python3.9/site-packages/jax/_src/numpy/lax_numpy.py", line 3486, in take_along_axis
    indices = _normalize_index(indices, axis_size)
  File "/tmp/tf2jax_test/lib/python3.9/site-packages/jax/_src/numpy/lax_numpy.py", line 3426, in _normalize_index
    return lax.select(
TypeError: a bytes-like object is required, not 'int'
```

The installed dependencies are as follows:

Package           Version
----------------- -------
absl-py           1.1.0
flatbuffers       2.0
jax               0.3.13
jaxlib            0.3.10
numpy             1.23.0
opt-einsum        3.3.0
pip               22.0.2
scipy             1.8.1
setuptools        59.6.0
typing_extensions 4.2.0

**Comment by jakevdp** [comment]
I suspect you're running into the numpy 1.23 incompatiblity (see https://github.com/google/jax/issues/11221) This is fixed on JAX head, but for now if you downgrade to numpy 1.22.4 it should fix the issue.

**Comment by shaobohou** [comment]
Is it fixed in jax 0.3.14? 
If so I could just wait for it to pop up on pypi? Not sure what the ETA is on that.

**Comment by jakevdp** [comment]
Yes, the fix will be in the next JAX release, which is in progress. Not sure precisely when it will land.

**Comment by shaobohou** [comment]
Just to clarify, you mean the fix will be in  0.3.15?

0.3.14 was released a few days ago, but on pypi it is still 0.3.13

**Comment by jakevdp** [comment]
I'm not sure what the number of the next release will be. We did tag 0.3.14 on github a few days ago, but due to some jaxlib build issues that release has not gone live.

**Comment by jakevdp** [comment]
I believe this is fixed as of jax v0.3.14
